packages/demo-nextjs/src/app/admin/(editor)/[...slug]/page.tsx

```diff
@@ -16,14 +18,18 @@ export default async function EditorPage(props: Props) {
   // Rollback to page collection when the item is missing
   if (!item) {
     item = collection;
-    collection = "page";
+    collection = CollectionTypes.page;
   }
 
-  const pagePreview = `${getConfig().host}/preview/${collection}/${item}`;
+  const pagePreview =
+    collection === CollectionTypes.page ? `${getConfig().host}/${item}` : `${getConfig().host}/${collection}/${item}`;
 
   try {
     const editorConfig = await getItemConfig({ item, collection });
-    return <Editor config={{ ...editorConfig, mode: Modes.page, pagePreview }} />;
+
+    const menu = [...(editorConfig?.menu ?? []), ...(await getMenuItems())];
+
+    return <Editor config={{ ...editorConfig, menu, mode: Modes.page, ...(pagePreview ? { pagePreview } : {}) }} />;
```

Comment:
Remove the `pagePreview` check, as it always exists and no longer needs validation.